### ISSUE TITLE
Support for Azure Database for MySQL Flexible Server

### ISSUE THREAD

**Issue by ms-lolo** [pulumi/pulumi-azure-native#1633]
## Hello!

- Vote on this issue by adding a 👍 reaction
- If you want to implement this feature, comment to let us know (we'll work with you on design, scheduling, etc.)

## Issue details

It doesn't seem like the https://www.pulumi.com/registry/packages/azure-native/api-docs/dbformysql/ API allows us to create flexible servers. This feels similar to #1455 but I'm not sure if the APIs for postgres and mysql are different enough that this one was missed. Nothing in the documentation is mentioning flexible servers so I'm assuming something is missing.

**Comment by jkodroff** [comment]
@ms-lolo Thanks for submitting this issue! We'll place it on our backlog.

**Comment by radirobi** [comment]
Hi @jkodroff,

do you have any estimation when this will be available?

**Comment by brunoroque06** [comment]
Flexible server works if one uses the `Pulumi.AzureNative.DBforMySQL.V20210501.Server` (C#).

However, what does not work then is configuring the server, because there is no `Pulumi.AzureNative.DBforMySQL.V20210501.Configuration`; there is only a `Pulumi.AzureNative.DBforMySQL.Configuration` which assumes the `Server` provided has the default API version (2017-12-01), which might not be the case. Then the server is not found:
```sh
error: Code="ResourceNotFound" Message="The Resource 'Microsoft.DBforMySQL/servers/mysql-somename' under resource group 'rg-somerg' was not found. For more details please go to https://aka.ms/ARMResourceNotFoundFix"
```

**Comment by lomholdt** [comment]
I'm hoping to get this fixed soon, as it seems Azure is sunsetting the single server setup and Flexible server will be the way going forward.

**Comment by jonas-lomholdt** [comment]
Is something like this open for contributions? I would probably be able to to give it a look if it's not planned for anytime soon :)

**Comment by danielrbradley** [comment]
@jonas-lomholdt we'd welcome support on solving this issue. On reading the issue it's not yet clear to me _why_ this functionality is missing. This would be the first stage to understand - where in the discovery process we're perhaps skipping the resource for some reason. Once we've established and confirmed a rational for why it's not there we'd need a proposal to address whatever the core issue is. Then once the solution plan has been checked over we'd move to a pull request.

To summarise the process:
1. Establish and document root cause
2. Propose solution
3. Address with code fix

It would be great if you can contribute to any of these steps! The priority here is step 1. Once step 1 is done, it's likely we'd be able to dedicate time internally to working towards steps 2 and 3 as the scope would then be a lot more clear.

**Comment by Madsmal** [comment]
Any update on this? Seems like Flexible Server is still not available and the single Server is soon to be deprecated.

**Comment by jonas-lomholdt** [comment]
I haven't had a chance to give it a look, I don't have the time at the moment.

**Comment by techgeek03** [comment]
@danielrbradley Any chance of prioritizing this issue as this looks more of a bug to me; and it is blocker for us.

From the official [API documentation](https://learn.microsoft.com/en-us/azure/templates/microsoft.dbformysql/flexibleservers?pivots=deployment-language-arm-template) we can see the API is there; however it is not been generated in the provider.
Checking https://github.com/pulumi/pulumi-azure-native/blob/master/provider/pkg/openapi/defaultResourcesState.go I can't find the flexible server for MySQL been added. Can this be the potential reason?

**Comment by mikhailshilkov** [comment]
An example of creating a MySQL flexible server and its configuration can be found here: https://github.com/pulumi/pulumi-azure-native/blob/master/examples/mysql/index.ts
